### PR TITLE
Improved the docs

### PR DESCRIPTION
**Pull request by mamazu**
* Fixing some typos
* Adding documentation:
   * We support `string[]` in doc types
   * We have a vscode extension
* Adding a note that the RPC is a legacy part of phpactor.
* Removed CoC as the only language server and made it more general.

### CODE REVIEW

File: doc/other/support.rst [hunk 1]
```diff
@@ -4,4 +4,4 @@ Support
 Need help? Create a Github issue or ping the Phpactor twitter account:
 
 -  `Github Issues <https://github.com/phpactor/phpactor/issues>`__
--  Twitter [@phpactor](https://twitter.com/phpactor)
+-  `Twitter [@phpactor] <https://twitter.com/phpactor>`__
```
Comment (dantleech) [inline doc/other/support.rst:7]: I'm posting more on https://phpc.social/@phpactor than Twitter currently, probably worth adding that now or later.

File: doc/usage/getting-started.rst [hunk 2]
```diff
@@ -3,19 +3,20 @@ Getting Started
 
 How you start depends on your editor.
 
-Currently Phpactor has most support when used over it's own RPC protocol, but
-an :ref:`increasing amount <lsp_support>` of support is offered from the
-:ref:`language_server`.
+Currently Phpactor has most support when used over it's own :ref:`legacy RPC protocol <rpc_protocol>`.
+Most of the functionality is implemented with a language server, which :ref:`those editors <lsp_support>`
+support. For this you need to run Phpactor as a Language Server.
```
Comment (dantleech) [inline doc/usage/getting-started.rst:8]: Can change this entire sentence:

```
Phpactor is now primarily a language server, but some additional functionality can still be accessed through the native VIM plugin
```

File: doc/usage/other-clients.rst [hunk 3]
```diff
@@ -17,3 +17,8 @@ Emacs Interface
 ---------------
 
 Github: `phpactor.el <https://github.com/emacs-php/phpactor.el>`_
+
+Visual Studio Code
+------------------
+
+Github: `vscode-phpactor <https://marketplace.visualstudio.com/items?itemName=williambrook.phpactor>`_
```
Comment (dantleech) [inline doc/usage/other-clients.rst:24]: this is not the official VS code extension (it's an extension to the leagcy RPC), the official LSP extension is: https://github.com/phpactor/vscode-phpactor

**Comment by mamazu** [comment]
I also added a fix to the documentation generator since it was removing white spaces when parsing.

**Pull request merged**